Tools/ardupilotwaf/ap_library.py

```diff
@@ -239,6 +239,20 @@ def __str__(self):
     def keyword(self):
         return 'Checking included headers'
 
+def custom_flags_check(tgen):
+    '''
+     check for tasks marked as having custom cpp or c flags
+     a library can do this by setting AP_LIB_EXTRA_CXXFLAGS and AP_LIB_EXTRA_CFLAGS
+    '''
+    if not tgen.name.startswith("objs/"):
+        return
+    libname = tgen.name[5:]
```

Comment:
Prefer defining `objs/` as a variable, then call length on it here in case it changes length.

Reply:
Something like this: 
```suggestion
    OBJS = "objs/"
    if not tgen.name.startswith(OBJS):
        return
    libname = tgen.name[len(OBJS):]
```

However, if `objs` is actually a path, then use pathlib and the `parts` operator. 
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parts

It's os-portable.